Bug 1203507 - [Spanish] AMO Page title larger than screen width

tsl143 opened the pull request.

tsl143 commented:
![after](https://cloud.githubusercontent.com/assets/5318732/12006549/eeacd780-ac00-11e5-8cb4-c0d2ac9c4a64.png)
![before](https://cloud.githubusercontent.com/assets/5318732/12006550/eeb25692-ac00-11e5-87c5-1b20ee5c96fd.png)

magopian commented on static/css/mobile/header.less:
```diff
@@ -181,9 +181,14 @@ header #auth-nav li {
+        text-overflow: ellipsis;
         text-transform: uppercase;
+        white-space: nowrap;    
+        width: 240px;
```
Is there a way to have that also work on other resolutions? ping @muffinresearch @tofumatt the CSS gurus ;)

muffinresearch replied:
Yep - I've noted the class to use in the other comment - but sounds like a plan to make that a goal that this should truncate long strings in both mobile and desktop.

muffinresearch commented:
Hi @tsl143 As mentioned in the bug we already look to have a helper class for this: see https://github.com/mozilla/olympia/blob/d8aef0b88a5fd20d4413a37807e9ae79c7602fa8/static/css/impala/lib.less#L135 was there a reason to not use that - it will save on duplication.

tsl143 commented:
Hi @muffinresearch I tried using just the class, but it seems we need to give width else the ellipsis effect doesnt show up.

muffinresearch commented:
@tsl143 that's fine - you should use the class and just add anything else needed where that's not enough.

tsl143 commented:
Hi @muffinresearch , I added ellipsis class to anchor and removed all repetitive definations from header.less. But on browser the ellipsis class is not defined ie., its defined in lib.less but not reflecting on the browser. Am I missing something again?

muffinresearch commented:
Sorry my bad looks like this is a mixin. You should find it will work if you add `.ellipsis();`  in the css as opposed to adding an actual class to the markup.

muffinresearch commented:
@tsl143 this is going to need rebasing against master. If you have a moment to do this and update to use the mixin, let me know when it's done and we can look to get this merged.

tsl143 commented:
@muffinresearch I am not sure if I understood what you mean by mixin? <sorry for being novice>

tsl143 commented:
@muffinresearch I got the mixin thing and updated the same :)

muffinresearch commented:
@tsl143 ok it's close now. I think it's better to use 100% here for the width otherwise you're going to always cause truncation even if there's more width available.

E.g: as it stands with width fixed at 240px:


<img width="216" alt="complementos_para_firefox" src="https://cloud.githubusercontent.com/assets/1514/12915077/6ba89a2a-cf20-11e5-8432-e675a1bbec76.png">

<img width="376" alt="complementos_para_firefox" src="https://cloud.githubusercontent.com/assets/1514/12915081/76e47e18-cf20-11e5-9a62-42cd0531f1cb.png">

And with 100%:

<img width="214" alt="complementos_para_firefox" src="https://cloud.githubusercontent.com/assets/1514/12915093/92e4a20a-cf20-11e5-9326-fc59e74bb51e.png">





<img width="367" alt="complementos_para_firefox" src="https://cloud.githubusercontent.com/assets/1514/12915089/877b1520-cf20-11e5-978e-98537344f9ef.png">

muffinresearch commented:
Actually scratch that, you can simply remove the width completely and it works correctly thanks to the `display: block`.

muffinresearch commented:
Once you've removed the width can you squash the commits [1] and I'll get this merged.

[1] http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

tsl143 commented:
@muffinresearch made the required changes, thanks for being patient :)

muffinresearch commented:
@tsl143 excellent thanks. Merged!
